pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -567,3 +578,28 @@ func hasUpgradeCommenced(c client.Client, uc *upgradev1alpha1.UpgradeConfig) (bo
 
 	return true, nil
 }
+
+func nodeDrained(c client.Client, uc *upgradev1alpha1.UpgradeConfig, logger logr.Logger) (bool, error) {
```

Comment:
We are not checking node/nodes drained here, because the node upgrade is one by one, better rename this to `hasStuckNode` maybe?

Reply:
Also please add a comment for the func

Reply:
the node is being drained during the upgrade process by the machine-config-operator 
https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update.go#L144

the node is marked as unschedulable when the drain started, and mark the start time in the spec.taints. And marked it as schedulable again after the drain finished.
The function here is checking that the node is marking as unschedulable longer than given duration.